packages/waku/src/router/index.ts

```diff
@@ -0,0 +1,22 @@
+import type { createPages } from './create-pages.js';
+import type { CollectPaths } from './inferred-router-types.js';
+type AnyRouter = ReturnType<typeof createPages>;
+
+type RouterSpec<Router extends AnyRouter> = Exclude<
+  Awaited<NonNullable<Router['DO_NOT_USE_ready']>>,
+  void
+>[number];
+
+// eslint-disable-next-line @typescript-eslint/no-empty-object-type
+export interface RouteConfig {
+  // routes to be overridden by users
+}
+
+export type PathsForRouter<Router extends AnyRouter> =
+  CollectPaths<RouterSpec<Router>> extends never
+    ? string
+    : CollectPaths<RouterSpec<Router>> & {};
+
+export type InferredPaths = RouteConfig extends { paths: infer UserPaths }
+  ? UserPaths
+  : string;
```

Comment:
I don't prefer `index.ts` here. Oh, it's required for `declare module 'waku/router' {` ?
If that's the case, can we only have L10-13 in this file?

Or, I'm fine with `declare module 'waku/router/client' {`. `'waku/router'` seems nicer.

Reply:
I can name the file anything you'd like. `declare module` just looks at the package.json exports and we set the filename there.

Reply:
would `src/router/router.ts` be ok?

The module is still `waku/router` externally.

Reply:
Anything other than `index.ts` is okay, but `router/router.ts` isn't nice.

Reply:
thoughts on `src/router/base.ts` or `src/router/export.ts`?

Or `src/router.ts`? it would seem weird to have a folder named `src/router` and also `src/router.ts` though.

Reply:
among them, src/router/base.ts sounds good. but then the name accepts more "base" logic in the file.

If this is all about typing, how about src/router/baseTypes.ts?

Reply:
changed to base-types.ts ✅